### PR TITLE
fix: Do not overwrite generated files with the same content

### LINKED ISSUE #14490: Extra Vite recompiles because generated files are always rewritten
### Description of the bug

If you make a change to the classpath for a running application, e.g. `touch target/classes/application.properties` then you will see that Vite recompiles several times during the Spring Boot restart cycle. This is because the files in `frontend/generated` are written again although there are no changes. Vite cannot know they have not changed, so it will recompile.

### Expected behavior

The files are only written if they have changed, so Vite does not have to do anything

### Minimal reproducible example

vaadin init hello
cd hello
mvn
touch classes/application.properties

### Versions

- Vaadin / Flow version: 23.2.0

### PR DESCRIPTION
**Pull request by Artur-**
Overwriting the files will cause the frontend tooling to recompile when there is no need to do that

Fixes #14490

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java [hunk 1]
```diff
@@ -66,18 +67,37 @@ public void execute() throws ExecutionFailedException {
...
+        FileUtils.writeStringToFile(file, content, UTF_8);
+    }
+
+    private static String getExistingFileContent(File generatedFile)
+            throws IOException {
+        if (!generatedFile.exists()) {
+            return null;
+        }
+        return FileUtils.readFileToString(generatedFile,
+                StandardCharsets.UTF_8);
```
Comment (mcollovati) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java:101]: nit: we can maybe use a static import for `UTF_8`, as we do in `writeIfChanged`
```suggestion
        return FileUtils.readFileToString(generatedFile, UTF_8);
```

**Review by mcollovati — COMMENTED** [review]
May we add some tests for `writeIfChanged`?

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java [hunk 2]
```diff
@@ -66,18 +67,37 @@ public void execute() throws ExecutionFailedException {
...
         }
     }
 
-    Logger log() {
-        return LoggerFactory.getLogger(getClass());
+    static Logger log() {
+        return LoggerFactory.getLogger(AbstractTaskClientGenerator.class);
+    }
+
+    static void writeIfChanged(File file, String content) throws IOException {
```
Comment (mshabarov) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java:82]: We need at least a unit test for this.

**Review by mshabarov — CHANGES_REQUESTED** [review]

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java (hunk 2 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java:82]: Added

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java (hunk 1 above)
Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/AbstractTaskClientGenerator.java]: Done

File: flow-server/src/test/java/com/vaadin/flow/server/frontend/AbstractTaskClientGeneratorTest.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.attribute.FileTime;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.nimbusds.jose.util.StandardCharset;
```
Comment (mshabarov) [inline flow-server/src/test/java/com/vaadin/flow/server/frontend/AbstractTaskClientGeneratorTest.java:13]: Should be `java.nio.charset.StandardCharsets` ?

Comment (Artur-) [reply flow-server/src/test/java/com/vaadin/flow/server/frontend/AbstractTaskClientGeneratorTest.java:13]: Fixed

**Review by mshabarov — DISMISSED** [review]

**Review by mshabarov — APPROVED** [review]

**Pull request merged**